src/order-book/api.ts

```diff
@@ -51,43 +58,132 @@ function cleanObjectFromUndefinedValues(obj: Record<string, string>): typeof obj
+ * const signer = provider.getSigner()
+ *
+ * const quoteRequest = {
+ *   sellToken: '0xb4fbf271143f4fbf7b91a5ded31805e42b2208d6', // WETH goerli
+ *   buyToken: '0x02abbdbaaa7b1bb64b5c878f7ac17f8dda169532', // GNO goerli
+ *   from: account,
+ *   receiver: account,
+ *   sellAmountBeforeFee: (0.4 * 10 ** 18).toString(), // 0.4 WETH
+ *   kind: OrderQuoteSide.kind.SELL,
+ * }
+ *
+ * const orderBookApi = new OrderBookApi({ chainId: SupportedChainId.GOERLI })
+ *
+ * async function main() {
+ *     const { quote } = await orderBookApi.getQuote(quoteRequest)
+ *
+ *     const orderSigningResult = await OrderSigningUtils.signOrder(quote, chainId, signer)
+ *
+ *     const orderId = await orderBookApi.sendOrder({ ...quote, ...orderSigningResult })
+ *
+ *     const order = await orderBookApi.getOrder(orderId)
+ *
+ *     const trades = await orderBookApi.getTrades({ orderId })
+ *
+ *     const orderCancellationSigningResult = await OrderSigningUtils.signOrderCancellations([orderId], chainId, signer)
+ *
+ *     const cancellationResult = await orderBookApi.sendSignedOrderCancellations({...orderCancellationSigningResult, orderUids: [orderId] })
+ *
+ *     console.log('Results: ', { orderId, order, trades, orderCancellationSigningResult, cancellationResult })
+ * }
+ * ```
+ *
+ * @see {@link Swagger documentation https://api.cow.fi/docs/#/}
+ * @see {@link OrderBook API https://github.com/cowprotocol/services}
+ */
 export class OrderBookApi {
   public context: ApiContext & RequestOptions
 
   private rateLimiter: RateLimiter
 
+  /**
+   * Creates a new instance of the CoW Protocol OrderBook API client.
+   * @param context - The API context to use. If not provided, the default context will be used.
+   */
   constructor(context: PartialApiContext & RequestOptions = {}) {
     this.context = { ...DEFAULT_COW_API_CONTEXT, ...context }
     this.rateLimiter = new RateLimiter(context.limiterOpts || DEFAULT_LIMITER_OPTIONS)
   }
 
+  /**
+   * Get the version of the API.
+   * @param contextOverride Optional context override for this request.
+   * @returns The version of the API.
+   * @see {@link https://api.cow.fi/docs/#/default/get_api_v1_version}
+   */
   getVersion(contextOverride: PartialApiContext = {}): Promise<string> {
     return this.fetch({ path: '/api/v1/version', method: 'GET' }, contextOverride)
   }
 
+  /**
+   * Get all the trades for either an `owner` **OR** `orderUid`.
+   *
+   * Given that an order *may* be partially fillable, it is possible that a discrete order (`orderUid`)
+   * may have *multiple* trades. Therefore, this method returns a list of trades, either for *all* the orders
+   * of a given `owner`, or for a discrete order (`orderUid`).
+   * @param request Either an `owner` or an `orderUid` **MUST** be specified.
+   * @param contextOverride Optional context override for this request.
+   * @returns A list of trades matching the request.
+   */
   getTrades(
     request: { owner?: Address; orderUid?: UID },
```

Comment:
A better way to handle this would be to do as i did in `getSolverCompetition`, just have two signature, one with owner, and the other with orderUid as a single parameter (method overloading)

this way is easier to use. However, feel free to ignore, since this would make a small breaking change

Reply:
Yeah, I saw the method overloading, and I'm not quite familiar with it in TS so I left it 😅

Reply:
okok, its fine, is actually unrelated to this PR, so probably good to keep out

I just mentioned cause this will make the assertions easier, and it will make it impossible to misused the service (i.e. pass the two parameters, or to not pass any of them)

Anyways, just a nit, lets keep it out...